src/storage.rs

```diff
@@ -189,28 +209,30 @@ impl<'a, S: Storage> DatabaseRef for StorageWrapper<'a, S> {
     type Error = S::Error;
 
     fn basic_ref(&self, address: Address) -> Result<Option<AccountInfo>, Self::Error> {
-        Ok(if let Some(basic) = self.0.basic(&address)? {
-            let code_hash = self.0.code_hash(&address)?;
-            let code = if let Some(code_hash) = &code_hash {
-                self.0.code_by_hash(code_hash)?.map(Bytecode::from)
-            } else {
-                None
-            };
-            Some(AccountInfo {
+        let basic = self.0.basic(&address)?;
+
+        Ok(basic.map(|basic| {
+            let code_hash = self.0.code_hash(&address).ok().flatten();
+            let code = code_hash
+                .as_ref()
+                .and_then(|hash| self.0.code_by_hash(hash).ok().flatten())
+                .map(|c| Bytecode::try_from(c).unwrap_or_default());
+
+            AccountInfo {
                 balance: basic.balance,
                 nonce: basic.nonce,
                 code_hash: code_hash.unwrap_or(KECCAK_EMPTY),
                 code,
-            })
-        } else {
-            None
-        })
+            }
+        }))
     }
 
     fn code_by_hash_ref(&self, code_hash: B256) -> Result<Bytecode, Self::Error> {
-        self.0
-            .code_by_hash(&code_hash)
-            .map(|code| code.map(Bytecode::from).unwrap_or_default())
+        self.0.code_by_hash(&code_hash).map(|code_option| {
+            code_option
+                .map(|c| Bytecode::try_from(c).unwrap_or_default())
```

Comment:
This falls back to a default bytecode on conversion error which seems quite dangerous. I suppose we'd need something like this:
```rust
enum StorageWrapperError<S: Storage> {
    StorageError(S::Error),
    InvalidBytecode(BytecodeConversionError),
}
```

Then `type Error = StorageWrapperError<S>;` for proper error-handling.